diff --git a/frontend/nova/coe_api/src/lib/py_coe_api.cpp b/frontend/nova/coe_api/src/lib/py_coe_api.cpp
index ef4c546091cc..cb4d26f95b15 100644
--- a/frontend/nova/coe_api/src/lib/py_coe_api.cpp
+++ b/frontend/nova/coe_api/src/lib/py_coe_api.cpp
@@ -1819,6 +1819,14 @@ PYBIND11_MODULE(rdu_engine, m) {
         .def_property_readonly(
             "shape",
             [](const RwLock<CoETensor> &t) { return py::tuple(py::cast(t.read(&profiler(), "CoETensor")->shape)); })
+        .def_property_readonly(
+            "dtype",
+            [](const RwLock<CoETensor> &t) {
+                auto layout = unwrap(t.read(&profiler(), "CoETensor")->logical_layout());
+                import_torch();
+                return pef_to_torch_dtype(layout.dtype);
+            },
+            py::doc("The torch.dtype of the underlying tensor."))
         .def("__getitem__",
              [](const RwLock<CoETensor> &tensor, py::object index) {
                  auto t = tensor.read(&profiler(), "CoETensor");
